autopilot/autopilot.go

```diff
@@ -9,188 +9,89 @@ import (
-	AddUploadingSectors(ctx context.Context, uID api.UploadID, root []types.Hash256) error
-	FinishUpload(ctx context.Context, uID api.UploadID) error
-	MarkPackedSlabsUploaded(ctx context.Context, slabs []api.UploadedPackedSlab) error
-	TrackUpload(ctx context.Context, uID api.UploadID) error
-	UpdateSlab(ctx context.Context, key object.EncryptionKey, sectors []api.UploadedSector) error
-
-	// locker
-	AcquireContract(ctx context.Context, fcid types.FileContractID, priority int, d time.Duration) (lockID uint64, err error)
-	KeepaliveContract(ctx context.Context, fcid types.FileContractID, lockID uint64, d time.Duration) (err error)
-	ReleaseContract(ctx context.Context, fcid types.FileContractID, lockID uint64) (err error)
-
-	// objects
-	Objects(ctx context.Context, prefix string, opts api.ListObjectOptions) (resp api.ObjectsResponse, err error)
-	RefreshHealth(ctx context.Context) error
-	Slab(ctx context.Context, key object.EncryptionKey) (object.Slab, error)
-	SlabsForMigration(ctx context.Context, healthCutoff float64, limit int) ([]api.UnhealthySlab, error)
-	DeleteHostSector(ctx context.Context, hk types.PublicKey, root types.Hash256) error
-	FetchPartialSlab(ctx context.Context, key object.EncryptionKey, offset, length uint32) ([]byte, error)
-
-	// scanner
-	ScanHost(ctx context.Context, hostKey types.PublicKey, timeout time.Duration) (resp api.HostScanResponse, err error)
-
-	// settings
-	GougingParams(ctx context.Context) (api.GougingParams, error)
 	GougingSettings(ctx context.Context) (gs api.GougingSettings, err error)
-	UploadSettings(ctx context.Context) (us api.UploadSettings, err error)
-
-	// syncer
-	SyncerPeers(ctx context.Context) (resp []string, err error)
-
-	// txpool
+	Hosts(ctx context.Context, opts api.HostOptions) ([]api.Host, error)
 	RecommendedFee(ctx context.Context) (types.Currency, error)
-	TransactionPool(ctx context.Context) (txns []types.Transaction, err error)
-
-	// wallet
+	ScanHost(ctx context.Context, hostKey types.PublicKey, timeout time.Duration) (api.HostScanResponse, error)
+	SyncerPeers(ctx context.Context) (resp []string, err error)
+	UploadSettings(ctx context.Context) (us api.UploadSettings, err error)
 	Wallet(ctx context.Context) (api.WalletResponse, error)
-	WalletPending(ctx context.Context) (resp []wallet.Event, err error)
-	WalletRedistribute(ctx context.Context, outputs int, amount types.Currency) (ids []types.TransactionID, err error)
 }
 
-type Autopilot struct {
-	alerts alerts.Alerter
-	bus    Bus
-	logger *zap.SugaredLogger
+type Autopilot interface {
+	Handler() http.Handler
+	Run()
+	Shutdown(context.Context) error
+	Trigger(forceScan bool) bool
+	Uptime() time.Duration
+}
+
+type autopilot struct {
+	b Bus
+	l *zap.SugaredLogger
 
-	c *contractor.Contractor
+	c contractor.Contractor
 	m migrator.Migrator
+	p pruner.Pruner
 	s scanner.Scanner
+	w wallet.Wallet
```

Comment:
Single letter fields aren't very descriptive.  I know we are probably not consistent across all of our code, but I'd rather use the full names like `logger`, `bus` etc. since they are short enough.